src/Confluent.Kafka/Producer.cs

```diff
@@ -670,8 +672,50 @@ internal Producer(ProducerBuilder<TKey, TValue> builder)
                 Librdkafka.conf_set_oauthbearer_token_refresh_cb(configPtr, oAuthBearerTokenRefreshCallbackDelegate);
             }
 
+            Action<SafeTopicConfigHandle, PartitionerDelegate> addPartitionerToTopicConfig = (topicConfigHandle, partitioner) =>
+            {
+                Librdkafka.topic_conf_set_partitioner_cb(topicConfigHandle.DangerousGetHandle(),
+                    (IntPtr rkt, IntPtr keydata, UIntPtr keylen, int partition_cnt, IntPtr rkt_opaque, IntPtr msg_opaque) =>
+                    {
+                        unsafe
+                        {
+                            var topicNamePtr = Librdkafka.topic_name(rkt);
+                            var topic = Util.Marshal.PtrToStringUTF8(topicNamePtr);
+                            var keyIsNull = keydata == IntPtr.Zero;
+                            var keyBytes = keyIsNull
+                                ? ReadOnlySpan<byte>.Empty
+                                : new ReadOnlySpan<byte>(keydata.ToPointer(), (int)keylen);
+                            return partitioner(topic, partition_cnt, keyBytes, keyIsNull);
+                        }
+                    });
+            };
+
+            // Configure the default custom partitioner.
```

Comment:
Is this part always performed after all other (global) config has been applied?

Reply:
yes - it's immediately before the handle is created.